librz/core/cbin.c

```diff
@@ -931,10 +938,46 @@ static char *construct_reloc_name(RZ_NONNULL RzBinReloc *reloc, RZ_NULLABLE cons
 	return rz_strbuf_drain(buf);
 }
 
-static void set_bin_relocs(RzCore *r, RzBinReloc *reloc, ut64 addr, Sdb **db, char **sdb_module) {
+static void reloc_set_flag(RzCore *r, RzBinReloc *reloc, const char *prefix, ut64 flag_addr) {
 	int bin_demangle = rz_config_get_i(r->config, "bin.demangle");
 	bool keep_lib = rz_config_get_i(r->config, "bin.demangle.libs");
 	const char *lang = rz_config_get(r->config, "bin.lang");
+	char *reloc_name = construct_reloc_name(reloc, NULL);
+	if (!reloc_name || !*reloc_name) {
+		free(reloc_name);
+		return;
+	}
+	char flagname[RZ_FLAG_NAME_SIZE];
+	if (r->bin->prefix) {
+		snprintf(flagname, sizeof(flagname), "%s.%s.%s", r->bin->prefix, prefix, reloc_name);
+	} else {
+		snprintf(flagname, sizeof(flagname), "%s.%s", prefix, reloc_name);
+	}
+	char *demname = NULL;
+	if (bin_demangle) {
+		demname = rz_bin_demangle(r->bin->cur, lang, flagname, flag_addr, keep_lib);
+		if (demname) {
+			snprintf(flagname, sizeof(flagname), "reloc.%s", demname);
+		}
+	}
+	rz_name_filter(flagname, 0, true);
+	RzFlagItem *existing = rz_flag_get(r->flags, flagname);
+	if (existing && existing->offset == flag_addr) {
+		// Mostly important for target flags.
+		// We don't want hundreds of reloc.target.<fcnname>.<xyz> flags at the same location
+		return;
+	}
```

Comment:
An alternative could be `reloc.target.<fcnname>.[a-z]` or in some case `reloc.target.<fcnname>.[acd]`

Reply:
What do you mean exactly? Usually you have many relocs pointing to the same target. Then we would like to have a single flag there called just `reloc.target.<fcnname>` without a suffix. (Suffixes might get added if there are multiple imported functions with the same name but then consequently different patch targets)

Reply:
simple case:
from
```
reloc.target.strcmp.0
reloc.target.strcmp.1
reloc.target.strcmp.2
reloc.target.strcmp.3
```
to
```
reloc.target.strcmp.[0-3]
```
complexe case (possible ?)
from
```
reloc.target.strcmp.0
reloc.target.strcmp.1
reloc.target.strcmp.3
```
to
```
reloc.target.strcmp.[013]
```

Reply:
Ah I see, that's a nice idea. But I think it would become too complicated here, in particular because the target flags than should also be perfectly in sync with the source flags.